### PR TITLE
Try to improve the readability and styling of matplotlibrc.template file

### PR DESCRIPTION
**Pull request by bingyao**
## PR Summary

This PR did NOT change any thing in `key` : `value` pairs, mainly change the formatting and styling in the comments, with blank lines or new lines. There are 3 commits, each for a set of modifications:

1. Commit f47cc5d: Update links, `http` -> `https`, update a outdated link, add few new links
2. Commit 7f7c0a0: Add a table of content at the top for quick skimming and navigation.
3. Commit a2c2555: Try to set and follow a few relaxed formatting and style conventions:

    ```
    ## Formatting and style conventions for this file:
    ##    - at least one whitesapce AROUND `:` to seperate key and val
    ##        * prefer one whitesapce except for colon alignment
    ##    - at least two whitesapces BEFORE `##` to seperate the key:val pair and
    ##      the trailing comments
    ##        * prefer two whitesapces except for block `##` alignment
    ##    - at least one whitesapce AFTER `##` to seperate the `##` marker and
    ##      the comment text
    ##        * prefer one whitesapce except for indentation (listing, etc.), in
    ##          which case, four more whitespaces are preferred
    ```
   I am not very sure about the third commit, whether it's necessary or will it pollute the `git blame` info. If this commit doesn't make sense, I can just drop the commit.

### CODE REVIEW

File: matplotlibrc.template [hunk 1]
```diff
@@ -55,61 +92,63 @@
...
+#interactive : False
+#toolbar     : toolbar2  ## None | toolbar2  ("classic" is deprecated)
+#timezone    : UTC       ## a pytz timezone string, e.g., US/Central or Europe/Paris
 
 ## Where your matplotlib data lives if you installed to a non-default
 ## location.  This is where the matplotlib fonts, bitmaps, etc reside
 #datapath : /home/jdhunter/mpldata
 
 
 #### LINES
```
Comment (timhoffm) [inline matplotlibrc.template]: The file is only plain text and has no formal format. Using sections in markdown-like style does not stand out enough. I suggest using something like
~~~
## *** LINES *****************************************************************
~~~
or even
~~~
## ***************************************************************************
## * LINES                                                                   *
## ***************************************************************************
~~~
for sectioning.

File: matplotlibrc.template [hunk 2]
```diff
@@ -1,46 +1,83 @@
 #### MATPLOTLIBRC FORMAT
 
 ## This is a sample matplotlib configuration file - you can find a copy
```
Comment (timhoffm) [inline matplotlibrc.template]: Just as a remark: The whole format description would better be placed at https://matplotlib.org/devdocs/tutorials/introductory/customizing.html#the-matplotlibrc-file. In the rc file itself, a 1-2 sentence description and the above link should be sufficient. This declutters the rc file. Additionally, the rendered docs are easier to read and search.

But I'm happy to leave that to another PR.

File: matplotlibrc.template (hunk 1 above)
Comment (bingyao) [reply matplotlibrc.template]: Great idea! And thanks for the reviewing!

File: matplotlibrc.template (hunk 2 above)
Comment (bingyao) [reply matplotlibrc.template]: I see. I can do this in another PR or with one or two more commits to declutters the RC file.

Comment (timhoffm) [reply matplotlibrc.template]: I think it's better done in a separate PR. This one is already quite large to review.

Comment (bingyao) [reply matplotlibrc.template]: Yes, it is. I will open another PR later on 😄

**Comment by tacaswell** [comment]
Thanks @bingyao !
